### PR TITLE
Component structure ex

### PR DESCRIPTION
**Pull request by MAlngaawy**
give an example of how to create the component in the library to be a reference.

- [x] initialize the component
- [x] initialize the component props types
- [x] create a story for the component

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
 
-yarn lint-staged
+yarn lint-staged --allow-empty
```
Comment (elhaw) [inline .husky/pre-commit:4]: why we need  `--allow-empty`

File: .storybook/main.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  stories: ['../stories/**/*.stories.@(ts|tsx|js|jsx)'],
+  stories: ['../stories/**/*.stories.@(ts|tsx|js|jsx)', '../src/**/*.stories.@(ts|tsx|js|jsx)'],
```
Comment (elhaw) [inline .storybook/main.js:2]: we will not add stories in this path `'../stories/**/*.stories.@(ts|tsx|js|jsx)'` as each stories will be in each  component folder

File: src/ExComponent/ExComponent.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import React from 'react';
+
+import { ExComponent } from './ExComponent';
+
+export default {
+  title: 'example component',
+  component: ExComponent,
+} as Meta<typeof ExComponent>;
+
+const Template: Story<typeof ExComponent> = (args) => <ExComponent {...args} />;
```
Comment (elhaw) [inline src/ExComponent/ExComponent.stories.tsx:11]: may you check this error

**Review by elhaw — CHANGES_REQUESTED** [review]

File: .husky/pre-commit (hunk 1 above)
Comment (MAlngaawy) [reply .husky/pre-commit:4]: It was for test .. I'm going to remove it

File: .storybook/main.js (hunk 2 above)
Comment (MAlngaawy) [reply .storybook/main.js:2]: I will remove it and the stories folder

File: src/ExComponent/ExComponent.stories.tsx (hunk 3 above)
Comment (MAlngaawy) [reply src/ExComponent/ExComponent.stories.tsx:11]: I have fix it but when commit it's add the () by default ! 
I will try to to fix it.

**Review by elhaw — APPROVED** [review]
LGTM.
Just do not forget to mention the eslint issue.

**Pull request merged**